thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,218 @@
+from .quay.common import get_configured_image_names
+from .quay.common import get_ps_s2i_image_names
+from .quay.common import get_image_containers
+from .quay.common import QUAY_TOKEN
+from .quay.common import QUAY_URL
+
+
+_LOGGER = logging.getLogger(__name__)
+
+_IMAGE_ANAYSIS_PRESCRIPTION_NAME = "thoth_image_analysis.yaml"
+_THOTH_IMAGE_ANALYSIS_WRAP = """\
+  - name: {prescription_name}BaseImageWrap
+    type: wrap
+    should_include:
+      adviser_pipeline: true
+      recommendation_types:
+      - latest
+      - performance
+      - stable
+      - testing
+      runtime_environments:
+        operating_systems:
+        - name: {os_name}
+          version: {os_version}
+        python_version: =={python_version}
+        base_images:
+          not:
+          - {image}
+    match:
+      state:
+        resolved_dependencies:
+        {resolved_dependencies}
+    run:
+      stack_info:
+      - type: INFO
+        message: >-
+          Found predictive stack image that can be used with these dependencies
+        link: {link}
+      advised_manifest_changes:
+      - file: .thoth.yaml
+        patch:
+        - name: base_image
+          value: {link}
+"""
+
+# TODO: Use Isis-API to identify categories or representative packages per topic?
+
+_REPRESENTATIVE_PACKAGES_ML = ["keras", "scikit-learn", "torch", "tensorflow"]
+
+_REPRESENTATIVE_PACKAGES_NLP = ["gensim", "nltk", "spacy", "transformers"]
+
+_REPRESENTATIVE_PACKAGES_CV = ["opencv-python", "pillow", "pytesseract", "torchvision"]
+
+_REPRESENTATIVE_PACKAGES = _REPRESENTATIVE_PACKAGES_ML + _REPRESENTATIVE_PACKAGES_NLP + _REPRESENTATIVE_PACKAGES_CV
+
+USER_API_HOST = os.environ["THOTH_USER_API_HOST"]
+
+
+def _get_latest_image_analyzed_info(image: str) -> Optional[Dict[str, Any]]:
+    """Get latest image analyzed information."""
+    url = f"http://{USER_API_HOST}/api/v1/container-images"
+    response = requests.get(url, params={"image_name": f"{QUAY_URL}/thoth-station/{image}"})
+
+    if response.status_code == 200:
+        results = response.json()
+        return results[0]
+    else:
+        return None
+
+
+def _get_requirement_files_from_image_analysis(
+    package_extract_document_id: str,
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """Get requirements files from image analysis result."""
+    url = f"http://{USER_API_HOST}/api/v1/analyze"
+    response = requests.get(url, params={"analysis_id": package_extract_document_id})
+
+    if response.status_code == 200:
+        document = response.json()
+
+        # Get result and requirements files
+        result = document["result"]
+        pipfile_dict = result["aicoe-ci"].get("requirements")
+        pipfile_lock_dict = result["aicoe-ci"].get("requirements_lock")
+
+        return pipfile_dict, pipfile_lock_dict
+
+    elif response.status_code == 404:
+        _LOGGER.warning(f"Document {package_extract_document_id} does not exist.")
+        return None, None
+
+    else:
```

Comment:
It might be a good idea to log status code here to see what has happened in logs. Here and in `_get_latest_image_analyzed_info` as well.

Reply:
Added!